src/main/java/us/ihmc/commons/thread/RepeatingTaskThread.java

```diff
@@ -331,14 +328,8 @@ public final void run()
    {
       while (running)
       {
-         synchronized (executionState)
-         {
-            if (executionState.getScheduled() == 0L)
-            {
-               executionState.waitForChange();
-               continue;
-            }
-         }
+         if (executionState.getScheduled() == 0L)
+            executionState.waitForChange();
```

Comment:
Pretty confident these should be in a synchronized block. Otherwise it's possible that the scheduled amount of executions is changed right between the condition check and the call to wait.

Reply:
reverted